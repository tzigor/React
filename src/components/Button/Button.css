.buttonSend {
    height: 30px;
    width: 130px;
    margin-top: 5px;
    border: 1px solid #A4A4A4;
    border-radius: 3px;
}