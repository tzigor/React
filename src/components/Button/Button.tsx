import './Button.css';
import { FC } from 'react';

interface ButtonProp {
    click?: () => void;
}

export const Button: FC<ButtonProp> = ({ click }) => {
    return (
        <button className="buttonSend" onClick={click}>
            Send message
        </button>
    );
}