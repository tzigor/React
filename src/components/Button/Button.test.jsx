import { Button } from './Button';
import { render, screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import '@testing-library/jest-dom';

describe('Button', () => {
  it('render component', () => {
    render(<Button>test</Button>);
  });

  it('render multiply buttons', () => {
    render(
      <>
        <Button>test</Button>
        <Button>test</Button>
      </>
    );
    expect(screen.queryAllByRole('button')).toHaveLength(2);
  });

  it('render with snapshot', () => {
    const { asFragment } = render(<Button label="test" />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('button async click', async () => {
    const mockHandler = jest.fn();
    render(<Button click={() => setTimeout(mockHandler, 1500)}>test</Button>);

    await userEvent.click(screen.getByText(/Send/));

    await waitFor(() => expect(mockHandler).toHaveBeenCalledTimes(1), {
      timeout: 1600,
    });

    // await screen.findByText('Clicked twice', { timeout: 1000 });
  });
});
