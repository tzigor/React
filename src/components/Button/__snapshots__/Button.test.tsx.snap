// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button render with snapshot 1`] = `
<DocumentFragment>
  <button
    class="buttonSend"
  >
    Send message
  </button>
</DocumentFragment>
`;
